libvast/vast/bloom_filter_synopsis.hpp

```diff
@@ -42,12 +43,24 @@ class bloom_filter_synopsis : public synopsis {
       default:
         return {};
       case relational_operator::equal:
+        // TODO: We should treat 'nil' as a normal value and
+        // hash it, so we can exclude synopsis where all values
+        // are non-nil.
+        if (caf::holds_alternative<view<caf::none_t>>(rhs))
+          return {};
+        if (!caf::holds_alternative<view<T>>(rhs))
+          return false;
         return bloom_filter_.lookup(caf::get<view<T>>(rhs));
       case relational_operator::in: {
         if (auto xs = caf::get_if<view<list>>(&rhs)) {
-          for (auto x : **xs)
+          for (auto x : **xs) {
```

Comment:
Sorry for being pedantic: the list itself can be `nil` as well, right?

Reply:
That's already covered by the `get_if`.